### ISSUE TITLE
Feature: Calendar tools

### ISSUE THREAD

**Issue by SallyMcGrath** [CodeYourFuture/curriculum#999]
This is a big concept ticket. If you're taking this to work on, break it down please.

We should be able to express our programme design as a calendar, like this spreadsheet does

<img width="1297" alt="Screenshot 2024-09-27 at 09 37 44" src="https://github.com/user-attachments/assets/2d2f58e3-d025-4369-b4c5-62954fb5b43b">

We should be able to derive this calendar from the [data/courses data](https://github.com/CodeYourFuture/curriculum/tree/main/org-cyf-theme/data/courses) eg name, frequency, duration. 

Let's try to do this.

Secondly, we should be able to provide some[ iCal integrations.](https://discourse.gohugo.io/t/calendar-ics-template/7137/4) Hugo has a calendar output format 

https://gohugo.io/templates/output-formats/#media-types

It seems like course dates should be calendar events, but I can also see opportunities for things like spaced repetition study aids, rhythm of the week reminders, etc. Please weigh in below.

**Comment by eastermukora** [comment]
indeed, for example the spaced repetition and rythm of the week reminders can be added as reminders. i suppose a `priority` part would include structuring calendar events as either daily/weekly. then the derivative implementations can be built atop. additionally, this can also be included in the .ics files, showing the event recurrence and reminders prior to events.
